### PR TITLE
Created game lobbies and Added Database Migrations for Users and Games

### PR DESCRIPTION
**Pull request by zhasa18**
# Implement Game Lobby Creation and Join with Join Codes

## Summary of Changes

1. **Game Lobby Creation and Join Codes:**
   - **Lobby Creation:**
     - Implemented functionality for users to create game lobbies.
     - Each new lobby is assigned a unique join code from user input.
     - Added a `GamesController` with actions for creating and managing games.
   - **Joining Games:**
     - Users can join existing lobbies using a join code.
     - Added a modal on the landing page to enter join codes.
     - Implemented validations to handle invalid or duplicate join codes.

2. **Database Migrations and Associations:**
   - **Foreign Keys and Associations:**
     - Created migrations to establish foreign keys between the `User` and `Game` models through a join table `game_users`.
   - **Model Updates:**
     - Updated the `Game` and `User` models to reflect the new associations.
     - Added validations to the `Game` model for join code uniqueness and presence.

3. **Landing Page Updates:**
   - **List of Joined Games:**
     - Updated the landing page to display a list of games the logged-in user has joined.
     - Each game entry shows the game name.
     - Included button to navigate to the game lobby.

## Testing

- **Lobby Creation and Joining:**
  - Tested creating new game lobbies to ensure unique join codes are generated.
  - Verified that users can join lobbies using valid join codes.

- **Database Integrity:**
  - Ran migrations and checked that foreign keys are correctly set up.
  - Tested the associations by having users join and leave games.
  - Ensured that deleting a user or game updates related records without issues.

- **Landing Page Functionality:**
  - Tested navigation links to ensure they direct to the correct game lobbies.
  
-**Landing Page Image**
image

-**Game Creation Modal Image**
image

-**Join  Game Image**
image

-**Temporary Game Page Image**
image

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -29,7 +29,10 @@ gem 'omniauth-google-oauth2'
 # Manage environment variable
 gem 'dotenv-rails', groups: [:development, :test]
 
-
+gem 'sassc-rails'
+gem 'bootstrap', '~> 5.3.3'
```
Comment (bjohnson66) [inline Gemfile:33]: This needed?

**Review by halimyu — APPROVED** [review]
Fantastic job! thank you @zhasa18

File: app/assets/images/landing_page.png
Comment (bjohnson66) [inline app/assets/images/landing_page.png:1]: nice image ( ͡° ͜ʖ ͡°)

**Review by bjohnson66 — APPROVED** [review]
Ran it locally, tests all pass with ~88% coverage
 (good)
 
 Things seem to work on local server

**Pull request merged**